### PR TITLE
Features/equate flows

### LINKED ISSUE #816: Missing constraint to relate flows
There is a constraint `equate_variables` in `oemof.solph.constraints` that allows to equate two variables or set them proportional with a factor. However, this does not work for flows, which are time-indexed.

I propose to implement a new constraint `equate_flows`. Before we start with this, I would like to get your feedback on some options for the api:

1.  Most straightforward: A constraint to relate two flows to each other
     `equate_flows(model, flow1, flow2, factor1=1, name=None)`.
2.  You might want to set a relation between not just two flows, but rather the sum of two groups. This could happen by keyword, e.g.: 
   `equate_flow_groups_by_keyword(model, keyword1, keyword1, factor1=1, name=None)`.
3.  More generally, one might want to relate other variables than flows, e.g. storage_content. This would look like this:
   `equate_time_indexed_var(model, var1, var2, factor1=1, name=None)`.

What are your thoughts, @oemof/oemof-solph? For a current modeling task we would probably most need no 2.

### PR DESCRIPTION
**Pull request by jnnr**
This PR fixes #816 by introducing a new constraint that allows to set flows or the sum of groups of flows equal or proportional by a constant factor.

The new constraint is located in oemof.solph.constraints.

You can either create the constraint by passing the names of the flows:

`equate_flows(model, [(fi_1, fo_1), [(fi_2, fo_2)]], [(fi_3, fo_3)])`

Or select them by a keyword:

`equate_flows_by_keyword(model, "fossile_flow", "renewable_flow")`

TODOs:
* Elaborate docstrings (similar to other constaints)
  * [x] Show equation (mathematical expression)
  * [x] List parameters and returns
  * [ ] What's new entry
  * [x] Add to api reference

### CODE REVIEW

**Comment by SabineHaas** [comment]
Thank you @jnnr for this nice feature! 

Concerning tests it would be nice to test `equate_flows_by_keyword` with more than one flow for each keyword.

I could test the constraint in our model as first use case if you like.

**Comment by SabineHaas** [comment]
@jnnr I have just realized: For using this new oemof.solph functionality we have to install the current branch (later oemof 5), of course. However, this will collide with `datapackage` (installed in `oemof.tabluar`) which still uses oemof v0.3 . 
I remember you saying that `datapackage` /  `oemof.tabluar` is planned to be upgraded on oemof v0.4, as well. Do you know when this would be?

There's also the following conflict. Does oemoflex work with the required pyomo versions of oemof-solph?
```The conflict is caused by:
    oemof-solph 0.4.5.dev0 depends on pyomo<5.7.3 and >=5.7.0
    oemoflex 0.0.0 depends on pyomo<5.6.9
```

**Review by p-snft — APPROVED** [review]
Reasonable addition - provided one is okay with non-component constraints.

File: src/oemof/solph/constraints/equate_flows.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+SPDX-FileCopyrightText: jnnr
+
+SPDX-License-Identifier: MIT
+
+"""
+
+from pyomo import environ as po
+
+
+def equate_flows(model, flows1, flows2, factor1=1, name="equate_flows"):
```
[the remark this answers is not in the document]
Comment (jnnr) [reply src/oemof/solph/constraints/equate_flows.py:14]: I will merge this and open a separate PR for renaming both.

**Pull request merged**